pysph/sph/gas_dynamics/tsph/scheme.py

```diff
@@ -0,0 +1,226 @@
+        as presented in Appendix F1 of [Hopkins2015]_ .
+
+        Notes
+        -----
+        Is this exactly in accordance with what is proposed in [Hopkins2015]_ ?
+            Not quite.
+
+        What is different then?
+            #. Adapting smoothing length using MPM [KP14]_ procedure from
+               :class:`SummationDensity
+               <pysph.sph.gas_dynamics.basic.SummationDensity>`. In this,
+               calculation of grad-h terms are changed to that specified for
+               this scheme.
+            #. Using the PEC integrator step. No individual
+               adaptive time-stepping.
+            #. Using :class:`Gaussian Kernel <pysph.base.kernels.Gaussian>`
+               by default instead of Cubic Spline with radius scale 1.
+
+        Tip: Reduce the number of points if particle penetration is
+        encountered. This has to be done while running
+        ``gas_dynamics.wc_blastwave`` and ``gas_dynamics.robert``.
+
+        Parameters
+        ----------
+        fluids: list
+            List of names of fluid particle arrays.
+        solids: list
+            List of names of solid particle arrays (or boundaries), currently
+            not supported
+        dim: int
+            Dimensionality of the problem.
+        gamma: float
+            :math:`\\gamma` for Equation of state.
+        hfact: float
+            :math:`h_{fact}` for smoothing length adaptivity, also referred to
+            as kernel_factor in other gas dynamics schemes.
+        beta : float, optional
+            :math:`\\beta` for artificial viscosity, by default 2.0
+        fkern : float, optional
+            :math:`f_{kern}`, Factor to scale smoothing length for equivalence
+            with classic kernel when using kernel with altered
+            `radius_scale` is being used, by default 1.
+        max_density_iterations : int, optional
+            Maximum number of iterations to run for one density step,
+            by default 250.
+        density_iteration_tolerance : float, optional
+            Maximum difference allowed in two successive density iterations,
+            by default 1e-3
+        has_ghosts : bool, optional
+            if ghost particles (either mirror or periodic) is used, by default
+            False
+        alphaav : float, optional
+            :math:`\\alpha_{av}` for artificial viscosity switch, by default
+            1.0
+        """
+
+        self.fluids = fluids
+        self.solids = solids
+        self.dim = dim
+        self.solver = None
+        self.gamma = gamma
+        self.beta = beta
+        self.hfact = hfact
+        self.density_iteration_tolerance = density_iteration_tolerance
+        self.max_density_iterations = max_density_iterations
+        self.has_ghosts = has_ghosts
+        self.fkern = fkern
+        self.alphaav = alphaav
+
+    def add_user_options(self, group):
+        group.add_argument("--alphaav", action="store", type=float,
```

Comment:
Suggestion: just think about the option names a bit and go with what is consistently used in the rest of the gas dynamics schemes.

Reply:
I did this because in the paper referred, it was ![equation](http://www.sciweavers.org/tex2img.php?eq=\alpha_{av}&fc=Black&im=svg&fs=10&ff=arev&edit=). Will change to `--alpha-max`.